bin/picca_coadd_zint.py

```diff
@@ -0,0 +1,458 @@
+    if not same_healpixs:
+        healpixs = np.array(list(xi.keys()))
+        xi = np.vstack([xi[healpix] for healpix in healpixs])
+        weights = np.vstack([weights[healpix] for healpix in healpixs])
+
+    # normalise xi by the weights
+    w = weights>0
+    xi[w] /= weights[w]
+
+    # normalize all other quantities by total weights
+    w = weights_total>0
+    r_par[w] /= weights_total[w]
+    r_trans[w] /= weights_total[w]
+    z[w] /= weights_total[w]
+
+    results = fitsio.FITS(fout, 'rw', clobber=True)
+
+    header = [
+    {
+        'name': 'RPMIN',
+        'value': r_par_min,
+        'comment': 'Minimum r-parallel [h^-1 Mpc]'
+    },
+    {
+        'name': 'RPMAX',
+        'value': r_par_max,
+        'comment': 'Maximum r-parallel [h^-1 Mpc]'
+    },
+    {
+        'name': 'RTMAX',
+        'value': r_trans_max,
+        'comment': 'Maximum r-transverse [h^-1 Mpc]'
+    },
+    {
+        'name': 'NP',
+        'value': headers_to_check_match_values['NP'],
+        'comment': 'Number of bins in r-parallel'
+    },
+    {
+        'name': 'NT',
+        'value': headers_to_check_match_values['NT'],
+        'comment': 'Number of bins in r-transverse'
+    },
+    {
+        'name': 'ZCUTMIN',
+        'value': z_cut_min,
+        'comment': 'Minimum redshift of pairs'
+    },
+    {
+        'name': 'ZCUTMAX',
+        'value': z_cut_max,
+        'comment': 'Maximum redshift of pairs'
+    },
+    {
+        'name': 'NSIDE',
+        'value': headers_to_check_match_values['NSIDE'],
+        'comment': 'Healpix nside'
+    },
+    {
+        'name': 'OMEGAM',
+        'value': headers_to_check_match_values['OMEGAM'],
+        'comment': 'Omega_matter(z=0) of fiducial LambdaCDM cosmology'
+    },
+    {
+        'name': 'OMEGAR',
+        'value': headers_to_check_match_values['OMEGAR'],
+        'comment': 'Omega_radiation(z=0) of fiducial LambdaCDM cosmology'
+    },
+    {
+        'name': 'OMEGAK',
+        'value': headers_to_check_match_values['OMEGAK'],
+        'comment': 'Omega_k(z=0) of fiducial LambdaCDM cosmology'
+    },
+    {
+        'name': 'WL',
+        'value': headers_to_check_match_values['WL'],
+        'comment': 'Equation of state of dark energy of fiducial LambdaCDM cosmology'
+    }
+    ]
+    results.write(
+        [r_par, r_trans, z, num_pairs],
+        names=['RP', 'RT', 'Z', 'NB'],
+        comment=['R-parallel', 'R-transverse', 'Redshift', 'Number of pairs'],
+        units=['h^-1 Mpc', 'h^-1 Mpc', '', ''],
+        header=header,
+        extname='ATTRI')
+
+    header2 = [{
+        'name': 'HLPXSCHM',
+        'value': 'RING',
+        'comment': 'Healpix scheme'
+    }]
+    results.write([healpixs, weights, xi],
+                  names=['HEALPID', 'WE', 'DA'],
+                  comment=['Healpix index', 'Sum of weight', 'Correlation'],
+                  header=header2,
+                  extname='COR')
+
+    results.close()
+
+    return
+
+def coadd_dmats(fi,fout):
```

Comment:
Just double checking. Why does this function does not check for `same_healpix` as is done in the previous function? Is it because the distortion matrix is already saved merging the information  from the different healpixs?

Reply:
Yes we don't measure the distortion matrix in HEALPix pixels, I guess because we're not interested in computing its variance via subsampling